pkg/controller/clusterdeployment/metrics.go

```diff
@@ -61,8 +63,22 @@ var (
 			Buckets: []float64{10, 30, 60, 300, 600, 1200, 1800},
 		},
 	)
+	metricProvisionFailedTerminal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "hive_cluster_deployments_provision_failed_terminal_total",
+		Help: "Counter incremented when a cluster provision has failed and won't be retried.",
+	},
+		[]string{"clusterpool_namespacedname"},
```

Comment:
The provision will not succeed. We set this when we've failed for the last time.

Reply:
Given how you're logging this metric, it would be deleted only when hive controller restarts - since it is not attached to anything else. So it should be cleared in the clusterpool controller for the relevant clusterpool

Reply:
I'm still not following this. I don't see why we should ever clear this metric. I definitely don't see why controller restart should be a significant event. Why should this metric behave any differently from e.g. [hive_cluster_deployments_installed_total](https://github.com/openshift/hive/pull/1524/files#diff-18db9c852df682803f7f52522e98b934cf23c39625767a593e4388fb367a0654R47-R52)?